contrib/juliac/juliac-buildscript.jl

```diff
@@ -27,6 +27,62 @@ if Base.JLOptions().trim != 0
     include(joinpath(@__DIR__, "juliac-trim-base.jl"))
 end
 
+const C_friendly_types = Base.IdSet{Any}([    # a few of these are redundant to make it easier to maintain
+    Int8, Int16, Int32, Int64, UInt8, UInt16, UInt32, UInt64, Float32, Float64, Bool,
+    Cvoid, Cint, Cshort, Clong, Cuint, Cushort, Culong, Cssize_t, Csize_t,
+    Cchar, Cwchar_t, Cstring, Cwstring,
+    RawFD,
+])
+
+function recursively_add_types!(types::Base.IdSet{DataType}, @nospecialize(T::DataType))
+    if T ∉ C_friendly_types
+        if T <: Ptr
+            return recursively_add_types!(types, T.parameters[1])
+        end
+        T.name.module === Core && error("invalid type for juliac: ", T) # exclude internals (they may change)
+        push!(types, T)
+    end
```

Comment:
```suggestion
    while T <: Ptr
        T = T.parameters[1] # unwrap Ptr{...}
    end
    if T.name.module === Core && T ∉ C_friendly_types
        error("invalid type for juliac: ", T) # exclude internals (they may change)
    end
    push!(types, T)
```

Mostly just to make the logic more direct, but there is a slight behavioral difference - if we add an `isbits struct` to `C_friendly_types`, this will make sure it still gets added to `types` for translation.